### DISCUSSION TITLE
Concern with merge process / governance

### DISCUSSION THREAD

**Discussion by aopsyche** [SynthstromAudible/DelugeFirmware#52] [General]
I see that several changes have already been merged to the codebase. This is great to see these bugfixes and formatting errors get adressed so quickly! However, i am a bit concerned that such changes were implemented without much community input or time elapsed. As far as i can tell, the process of merger arbitration hasnt been adequately discussed, standards and goals have not finished being established, and developers at present cannot be confident that their current work will remain compatible.

To @jamiefaye and @litui: i dont mean to imply that i do not trust your judgement, only that i dont think trust will always be a sufficient solution for arbitrating when and what code gets merged.

Some of these concerns could be adressed by having multiple community branches as a buffer between contribution and inclusion into the main community release (such as experimental or weekly/nightly builds). Some of these concerns could be addressed by democratizing either the arbitration process itself or at least democratizing the creation of the system we will use to arbitrate. I understand that not all decisions can be efficiently distributed to the whole community, especially while things are just getting off the ground, but i think there are immediate community wide benefits for taking democratic consideration in some degree.

I am only a hobbyist programmer and have very little experience with open source projects. I will not likely contribute code to this project within the next few months, and thus have very little stake in the issue (except as an end user who desires good output from this project). I'm relatively ignorant about these issues, if indeed my concerns have foundation at all. I do have experience with methods of governance, however, and my immediate thought is that the current one (or lack thereof) will not remain satisfactory for the community. At the very least, i think developers would appreciate more standardization of the review and merger process, despite how the merger decision is made.

If i'm completely off base: my apologies, and i'm excited to learn more! Pardon my ignorance, but i have had growing concern over these issues for the last week, and would be remiss if i didnt bring up this concern publicly in the case that anyone else feels similar.

**Comment by litui** [comment]
Thanks for the thought you've clearly put into this and for taking the time to express your concerns. Honestly I agree, and in the Discord at least I've been suggesting that both a governance model and a clear direction for the `community` repo need to happen sooner rather than later as contributors venture into discussion of potentially breaking or unstable features, feature selection mechanisms, and workflow changes.

(For reference, I've also done my time as a policy & governance wonk beyond my work in the technical sphere so I'm with you on the need for defined process and structure even if I don't think I'm the one who should be proposing it).

Even as someone not necessarily contributing code, consider starting or contributing to a draft pull request relating to the GOVERNANCE.md document and/or CONTRIBUTING.md document. Your suggestions and contributions are welcome and both this thread and a draft pull request could be just the ticket to getting some of these outstanding hurdles resolved.

**Reply by litui** [reply]
To follow up @aopsyche, PR #56 has been opened by one of the community members. Please have a look and contribute your advice!

**Reply by aopsyche** [reply]
Awesome! Yes i have seen the new relevant PR and i'm excited to see it develop. I appreciate your encouragement, and will try to contribute where i can!
